src/libraries/System.Memory/tests/Span/IndexOfAny.char.cs

```diff
@@ -772,5 +773,70 @@ public static void MakeSureNoChecksGoOutOfRangeMany_Char()
                 Assert.Equal(-1, index);
             }
         }
+
+        [Fact]
+        public static void TestIndexOfAny_RandomInputs_Char()
+        {
+            IndexOfAnyCharTestHelper.TestRandomInputs(
+                expected: IndexOfAnyReferenceImpl,
+                actual: (searchSpace, values) => searchSpace.IndexOfAny(values));
+
+            static int IndexOfAnyReferenceImpl(ReadOnlySpan<char> searchSpace, ReadOnlySpan<char> values)
+            {
+                for (int i = 0; i < searchSpace.Length; i++)
+                {
+                    if (values.Contains(searchSpace[i]))
+                    {
+                        return i;
+                    }
+                }
+
+                return -1;
+            }
+        }
+    }
+
+    public static class IndexOfAnyCharTestHelper
+    {
+        public delegate int IndexOfAnySearchDelegate(ReadOnlySpan<char> searchSpace, ReadOnlySpan<char> values);
+
+        public static void TestRandomInputs(IndexOfAnySearchDelegate expected, IndexOfAnySearchDelegate actual)
+        {
+            TestRandomInputs(127, expected, actual);
+            TestRandomInputs(char.MaxValue, expected, actual);
+        }
+
+        private static void TestRandomInputs(int maxNeedleValue, IndexOfAnySearchDelegate expected, IndexOfAnySearchDelegate actual)
+        {
+            maxNeedleValue++;
+
+            Span<char> needleSpace = stackalloc char[8];
+            Span<char> haystackSpace = stackalloc char[40];
+            var rng = new Random();
```

Comment:
To make this reproducable, we should include a seed here.

Reply:
If I remember correctly, I removed it because the RNG part represents a significant portion of the test execution time, with the non-explicit-seed ctor using the faster impl AFAIK.

For reproducibility, I made sure to emit the exact inputs that failed. Does that address your concerns here?

Reply:
> Does that address your concerns here?

Only partially. There are two concerns:
1. That a test could fail and you not know why.
2. That two runs of the same test suite might be non-deterministic.

Outputting the exact inputs addresses (1) but not (2). And (2) in general contributes to test flakiness.

If this test is about randomly stressing the implementation, then I don't believe it belongs in the functional test suite.  If it's about using pseudo-randomness to try out thousands of inputs without having to manually code them all and be creative enough to ensure a reasonable spread of inputs, then it should be done in a way that runs the same tests every time, which means using a seed.

Reply:
I changed it to waste less time in Random and use a constant seed.

It was extremely useful when working on the initial implementation to get rid of all the edge-case bugs. Some were so odd that I don't believe we would have come up with test cases for otherwise.

I increased the number of iterations so it now takes about a second of CPU time to run through them, so I moved these to outerloop.
Outerloop already takes ~4 minutes on my machine so I'm gonna assume that's acceptable.